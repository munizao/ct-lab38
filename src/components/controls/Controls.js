import React from 'react';
import PropTypes from 'prop-types';
import styles from './Controls.css';

const Controls = ({ actions, dispatch }) => {
  console.log(actions);
  return (
    <section className={styles.Controls}>
      {actions.map(({ action, text, count }) => (
        <button key={text} onClick={() => dispatch(action())}>
          {text}
          {!!count && `- ${count}`}
        </button>
      ))}
    </section>
  );
};

Controls.propTypes = {
  actions: PropTypes.arrayOf(PropTypes.shape({
    action: PropTypes.func.isRequired,
    text: PropTypes.string,
    count: PropTypes.number
  })).isRequired,
  dispatch: PropTypes.func.isRequired
};

export default Controls;
