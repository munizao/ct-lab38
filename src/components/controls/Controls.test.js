import React from 'react';
import { shallow } from 'enzyme';
import Controls from './Controls';

describe('Controls component', () => {

  it('renders Controls', () => {
    const wrapper = shallow(<Controls />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
