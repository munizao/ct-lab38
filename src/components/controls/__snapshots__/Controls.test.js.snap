// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls component renders Controls 1`] = `
<section
  className="Controls"
>
  <Button
    actionObj={
      Object {
        "action": [Function],
        "stateName": "coffees",
        "text": "Drink Coffee",
      }
    }
    key="Drink Coffee"
  />
  <Button
    actionObj={
      Object {
        "action": [Function],
        "stateName": "snacks",
        "text": "Snack",
      }
    }
    key="Snack"
  />
  <Button
    actionObj={
      Object {
        "action": [Function],
        "stateName": "naps",
        "text": "Nap",
      }
    }
    key="Nap"
  />
  <Button
    actionObj={
      Object {
        "action": [Function],
        "stateName": "studies",
        "text": "Study",
      }
    }
    key="Study"
  />
</section>
`;
